rpcs3/Emu/RSX/RSXThread.cpp

```diff
@@ -2414,12 +2417,20 @@ namespace rsx
 			else
 			{
 				// TODO: Fix this
-				u32 ea = address >> 20, io = RSXIOMem.io[ea];
+				u32 ea = address >> 20, io = iomap_table.io[ea];
 
-				for (const u32 end = ea + (size >> 20); ea < end;)
+				if (io + 1)
 				{
-					offsetTable.ioAddress[ea++] = 0xFFFF;
-					offsetTable.eaAddress[io++] = 0xFFFF;
+					io >>= 20;
+
+					const auto cfg = g_fxo->get<gcm_config>();
+					std::lock_guard lock(cfg->gcmio_mutex);
+
+					for (const u32 end = ea + (size >> 20); ea < end;)
+					{
+						cfg->offsetTable.ioAddress[ea++] = 0xFFFF;
```

Comment:
These constants are 16-bit, but the table entries were modifed to be 32-bit. (0xFFFF + 1) != 0.

Reply:
Also, just use the constants in stdint for these (UINT32_MAX, UINT16_MAX, etc).
Also, if these range clears happen more than in one place, consider making it a proper method in the iomap table class.

Reply:
oh, those are gcm tables, they still use 16-bit entries.

Reply:
Then I have to ask why maintain two map table classes. Might as well unify them.

Reply:
hmm does not really work like that, gcm table is in vm memory while rsx is not.
Games can damage (with writes) gcm table but not the rsx one.

Reply:
So the PS3 has 2 separate tables as well? The vm memory limitation is easily worked around by using a template.

Reply:
I suppose that can be considered outside PR scope anyway. Don't worry too much about it.

Reply:
yea it has two tables.
idk about using a template, with this pr they have quite a few new differences.